### PR TITLE
Improve error handling for '/containerLogs' API in kubelet.

### PR DESCRIPTION
**Pull request by vishh**
#3391

### CODE REVIEW

**Comment by vishh** [comment]
@dchen1107: I see that many of the kubelet APIs can follow a input validation pattern similar to this PR. WDYT?

**Comment by dchen1107** [comment]
LGTM, will merge tomorrow morning.

**Comment by smarterclayton** [comment]
This broke how we fetch logs for containers that are stopped - I don't think our intent was to not return logs for stopped containers, just make it deterministic.

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1415,19 +1415,33 @@ func (kl *Kubelet) GetDockerVersion() ([]uint, error) {
...
+	if podStatus.Phase != api.PodRunning {
+		return fmt.Errorf("pod %q is not in 'Running' state - State: %q", podFullName, podStatus.Phase)
 	}
-	return dockertools.GetKubeletDockerContainerLogs(kl.dockerClient, dockerContainer.ID, tail, follow, stdout, stderr)
+	exists := false
+	dockerContainerID := ""
+	for cName, cStatus := range podStatus.Info {
+		if containerName == cName {
+			exists = true
+			if !cStatus.Ready {
```
Comment (smarterclayton) [inline pkg/kubelet/kubelet.go]: Specifically, this check is too aggressive, it means you cannot return logs for a container that ran to completion and succeeded (a RestartPolicyNever)

**Pull request merged**